graphprot/NeuralNet.py

```diff
@@ -236,9 +251,42 @@ def train(self, nepoch=1, validate=False, plot=False):
                     'eval', self.threshold).ACC
                 self.valid_acc.append(_val_acc)
 
+                # Print loss and accuracy (validation set)
                 self.print_epoch_data(
                     'valid', epoch, _val_loss, _val_acc, t)
 
+                # save the best model (i.e. lowest loss value on validation data)
+                if save_model == 'best' :
+                    
+                    if min(self.valid_loss) == _val_loss :
+                        self.save_model(filename='t{}_y{}_b{}_e{}_lr{}.pth.tar'.format(
+                            self.task, self.target, str(self.batch_size), str(nepoch), str(self.lr)))
+
+            else :
+                # if no validation set, saves the best performing model on the traing set 
+                if save_model == 'best' :
+                    if min(self.train_loss) == _train_loss :
+                        print ('WARNING: The training set is used both for learning and model selection.')
+                        print('this may lead to training set data overfitting.')
+                        print('We advice you to use an external validation set.')
+                        self.save_model(filename='t{}_y{}_b{}_e{}_lr{}.pth.tar'.format(
+                            self.task, self.target, str(self.batch_size), str(nepoch), str(self.lr)))
+
+            # Save epoch data
+            if (save_epoch == 'all') or (epoch == nepoch) :
+                self._export_epoch_hdf5(epoch, self.data)
+            
+            elif (save_epoch == 'intermediate') and (epoch%5 == 0) :
```

Comment:
you could add `save_every=n` as named argument of the function so that we can decide how often we save

Reply:
Done, defaults is set to 5